core/src/main/java/org/renjin/sexp/Environment.java

```diff
@@ -461,9 +495,9 @@ public void accept(SexpVisitor visitor) {
   }
 
   public SEXP getVariable(Context context, Symbol symbol) {
-    assert ( context != null);
+    assert ( context != null );
     if(activeBindings != null && activeBindings.containsKey(symbol)) {
-      return Evaluation.eval(context, getActiveBinding(symbol), parent, ExpressionVector.EMPTY);
+      return context.evaluate(activeBindings.get(symbol));

```

Comment:
This doesn't look correct: an SEXP of type closure is treated as a constant, so evaluating a closure will just return the closure. You need to construct and evaluate a **function call**.